### DISCUSSION TITLE
We've lost some metadata? `2.3.x`

### DISCUSSION THREAD

**Discussion by gAlleb** [savonet/liquidsoap#4072] [Q&A] [answered]
Just noticed, that new 2.3.x is missing `on_air_timestamp` and `on_air` metadata. Or am I wrong?

Just tested with RR both on opam and Alpine. Can't get values of that fields.

**Comment by vitoyucepi** [comment, marked as answer]
Since #4039, but I don't know why.

**Reply by gAlleb** [reply]
True, maybe convert into issue cause `on_air` is crucial for some _nowplaying api_ implementations :)

**Comment by gAlleb** [comment]
created https://github.com/savonet/liquidsoap/issues/4073

**Comment by toots** [comment]
Hi all.

I cleaned up the `request` imeplemtation as part of `2.3.x`. That code was super old and wasn't reflecting how we do things now.

In particular, `on_air` was a proper state of requests, beside `idle`, `resolving`, `resolved` and `destroyed`. This complicated the code a lot and did not reflect on the fact that a request can be used by multiple sources.

I understand the need to track what's on air at all time. I'll follow-up with some suggestions for adding it back in the feature/bug report.
